plugin/https/README.md

```diff
@@ -0,0 +1,103 @@
+# https
+
+## Name
+
+*https* - facilitates proxying DNS messages to upstream resolvers via DoH protocol.
+
+## Description
+
+The *https* plugin supports DoH.
+
+This plugin can only be used once per Server Block.
+
+## Syntax
+
+In its most basic form:
+
+~~~
+https FROM TO...
+~~~
+
+* **FROM** is the base domain to match for the request to be proxied.
+* **TO...** are the destination endpoints to proxy to. The number of upstreams is
+  limited to 15.
+
+Multiple upstreams are randomized (see `policy`) on first use. When a proxy returns an error
+the next upstream in the list is tried.
+
+Extra knobs are available with an expanded syntax:
+
+~~~
+https FROM TO... {
+    except IGNORED_NAMES...
+    tls CERT KEY CA
+    tls_servername NAME
+    policy random|round_robin|sequential
+}
+~~~
+
+* **FROM** and **TO...** as above.
+* **IGNORED_NAMES** in `except` is a space-separated list of domains to exclude from proxying.
+  Requests that match none of these names will be passed through.
+* `tls` **CERT** **KEY** **CA** define the TLS properties for TLS connection. From 0 to 3 arguments can be
+  provided with the meaning as described below
+
+  * `tls` - no client authentication is used, and the system CAs are used to verify the server certificate (by default)
+  * `tls` **CA** - no client authentication is used, and the file CA is used to verify the server certificate
+  * `tls` **CERT** **KEY** - client authentication is used with the specified cert/key pair.
+    The server certificate is verified with the system CAs
+  * `tls` **CERT** **KEY**  **CA** - client authentication is used with the specified cert/key pair.
+    The server certificate is verified using the specified CA file
+
+* `policy` specifies the policy to use for selecting upstream servers. The default is `random`.
+
+Also note the TLS config is "global" for the whole https proxy if you need a different
+`tls` for different upstreams you're out of luck.
+
+## Metrics
+
+If monitoring is enabled (via the *prometheus* plugin) then the following metric are exported:
+
+* `coredns_https_request_duration_seconds{to}` - duration per upstream interaction.
+* `coredns_https_requests_total{to}` - query count per upstream.
+* `coredns_https_responses_total{to, rcode}` - count of RCODEs per upstream.
+  and we are randomly (this always uses the `random` policy) spraying to an upstream.
```

Comment:
where is the code for these metrics?

Reply:
this [commit](https://github.com/coredns/coredns/pull/4583/commits/ec1fd0a2873c9a372c2ce952dd59e79397823a1a) clearly demonstrates the flexibility of using Golang interfaces instead of hardcoding all logic in main DoH client which would otherwise make it [god object](https://en.wikipedia.org/wiki/God_object).